File: stu_spring/webflux/auth/src/main/java/com/webflux/auth/global/error/exception/ErrorCode.java
package com.webflux.auth.global.error.exception;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum ErrorCode {

    USER_NOT_FOUND("User Not Found Exception", 404, "User Error"),
    USER_ALREADY_EXIST("User Id Already Exist Exception", 409, "User Error"),
    QUERY_PARAMETER_NOTFOUND("Query Parameter Not Found Exception", 400, "Blog Error"),
    BLOG_NOT_FOUND("Blog NotFound Exception", 404, "Blog Error"),
    INVALID_TOKEN("Invalid Token Exception", 401, "Token Error");

    private final String message;
    private final int status;
    private final String error;
}

File: kotlin/kotlin_basic/src/main/kotlin/com/kotlin/test/kotlintest/global/exception/enums/ErrorCode.kt
package com.kotlin.test.kotlintest.global.exception.enums

enum class ErrorCode(
    val status: Int,
    val cause: String
) {
    USER_NOT_FOUND(404, "User Not Found"),
    PASSWORD_NOT_MATCH(404, "Password Not Match"),
    USER_ALREADY_EXIST(409, "User Already Exist"),
    INVALID_TOKEN(401, "Invalid Token"),
    TOKEN_EXPIRED(401, "Token Expired"),
    POST_NOT_ACCESSIBLE(403, "Not Accessible"),
    POST_NOT_FOUND(404, "Post Not Found"),
    METHOD_ARGUMENT(400, "Invalid Method Argument"),
    NO_HANDLER(404, "No Controller Handler"),
    UNEXPECTED_EXCEPTION(500, "Unexpected Exception")
}